packages/experience-builder-sdk/src/hooks/useDetectMode.tsx

```diff
@@ -28,29 +30,45 @@ export const useDetectEditorMode = ({ isClientSide = false }: UseDetectEditorMod
 
       if (eventData.eventType === INCOMING_EVENTS.RequestEditorMode) {
         setIsEditorMode(true);
-        receivedMessage.current = true;
+        receivedEditorModeMessage.current = true;
         if (typeof window !== 'undefined') {
           //Once we definitely know that we are in editor mode, we set this flag so future postMessage connect calls are not made
           if (!window.__EB__) {
             window.__EB__ = {};
           }
           window.__EB__.isEditorMode = true;
-          window.removeEventListener('message', onMessage);
         }
+      } else if (eventData.eventType === INCOMING_EVENTS.RequestReadOnlyMode) {
+        setIsReadOnlyMode(true);
```

Comment:
```suggestion
        setIsReadOnlyMode(true);
        setIsEditorMode(false);
```

To be safe, it would ensure that the other mode is turned off (same in line 32)

Reply:
Totally agree, this was something that @jsdalton also called out via Slack.  I've since updated to the code so that only 1 mode can be enabled at one time.

At any given time, mode will be either `MODE.READ_ONLY` `MODE.EDITOR` or `MODE.NONE`.